lib/cpp/src/thrift/transport/THeaderTransport.cpp

```diff
@@ -375,7 +383,12 @@ void THeaderTransport::resetProtocol() {
 }
 
 uint32_t THeaderTransport::getWriteBytes() {
-  return wBase_ - wBuf_.get();
+  ptrdiff_t wb = wBase_ - wBuf_.get();
+  if (wb > std::numeric_limits<uint32_t>().max()) {
+    throw TTransportException(TTransportException::CORRUPTED_DATA,
+                              "write size is unreasonable");
+  }
```

Comment:
Unfortunately using a template like this caused build warnings to occur when the inputs are not the same signed-ness.  I think these bounds checks need to be done with eyes wide open, and need to be specific in each case in order to avoid compiler warnings.